lib/post.js

```diff
@@ -0,0 +1,21 @@
+import config from "../config";
+
+const STRAPI_URL = config.STRAPI_URL;
+
+// Helper function to GET the posts from Strapi
+async function fetchPostsApi() {
+  const requestUrl = `${STRAPI_URL}/api/posts?populate=*`;
+  const response = await fetch(requestUrl);
+  const data = response.json();
+  return await data;
+}
+
+// Helper function to GET a single post from Strapi
+async function fetchPostApi(slug) {
+  const requestUrl = `${STRAPI_URL}/api/posts/${slug}?populate=*`;
+  const response = await fetch(requestUrl);
+  const data = response.json();
+  return await data;
```

Comment:
Make common method for fetching data from the given URL and pass just URL from `fetchPostsApi` and `fetchPostApi`